core/src/main/java/hudson/slaves/OfflineCause.java

```diff
@@ -54,17 +55,19 @@
      */
     public static class SimpleOfflineCause extends OfflineCause {
         public final Localizable description;
+        public final Date timestamp;
```

Comment:
I've made the change and it seems to work locally. 
A beginner's question, if I may: how to run the built-in selftests on my local machine, preferably from a command line, before committing to this site? The links I've found haven't answered this. I'm new at this, and appreciate your answer.